src/WebcamInput.jl

```diff
@@ -0,0 +1,522 @@
+	}
+	.controls {
+		grid-column-start: 2;
+		grid-row-start: 3;
+		display: flex;
+		align-items: center;
+		justify-content: center;
+		margin:0.2em;
+		z-index: 2;
+	}
+	.controls > button {
+		margin-left: 0.2em;
+	}
+	.ionic {
+		display: inline-block;
+		width: 1em;
+		height: 1em;
+		line-height: 1em;
+		margin-bottom: -2px;
+	}
+	.ionic-cam {
+		background-image: url("https://unpkg.com/ionicons@5.5.2/dist/svg/camera-outline.svg");
+	}
+</style>""")
+
+# ╔═╡ 063bba88-ef00-4b5b-b91c-14b497da85c1
+begin 
+	a = 4
+	function Base.show(io::IO, ::MIME"text/html", webcam::WebcamInput)
+	id = string("id-", webcam.uuid)
+	wwidth = webcam.width
+	wheight = webcam.height
+	aspectRatio = webcam.aspectRatio
+	webcam.help && @info "Note that the Webcam only works in localhost or https:// contexts!\n Width, height and aspect ratio are requested to the User Agent (browser) but may not be available, so the UA will try to do its best!"
+	show(io, MIME("text/html"), @htl("""
+	<div class="webcam">
+		$(css)
+		<script id="$(id)">
+	if (this) return this;
+
+	const checkIfHasMedia = () => {
+		return !!(
+			navigator.mediaDevices &&
+			navigator.mediaDevices.getUserMedia
+		)
+	};
+	const hasMedia = checkIfHasMedia()
+	
+	const video = html`<video autoplay></video>`
+	const canvas = html`<canvas style="display: none"></canvas>`
+	const start_video_ctl = html`<button>
+		start <span class="ionic ionic-cam" />
+		</button>`;
+
+	const capture_ctl = html`<button>
+		capture <span class="ionic ionic-cam" />
+		</button>`;
+
+	const stop_video_ctl = html`<button>
+		stop <span class="ionic ionic-cam" />
+		</button>`;
+
+	const state = {
+		initialized: false,
+		streaming: false,
+		error: false,
+		stream: null,
+		height: 0,
+		width: 0,
+	};
+
+	const tryInitVideo = () => {
+		const constraints = {
+  			video: {
+				width: {ideal: $wwidth},
+	  			height: {ideal: $wheight},
+			  	aspectRatio: {ideal: $aspectRatio}
+			}
+		};
+		console.log(constraints)
+		navigator.mediaDevices
+			.getUserMedia(constraints)
+			.then((stream) => {
+				state.stream = stream;
+				let {width, height} = stream.getTracks()[0].getSettings();
+				state.width = width
+				state.height = height;
+				video.srcObject = stream;
+			}).catch(err => {
+				state.stream = null;
+				state.width = 0;
+				state.height = 0;
+				state.error = err;
+			});
+	}
+	const closeCamera = () => {
+		video.srcObject = null;
+		
+		if(state.stream)
+			state.stream.getTracks().forEach(track => {
+				track.readyState == "live" && track.stop()
+			})
+
+		state.stream = null;
+		state.width = 0;
+		state.height = 0;
+	}
+
+	const capture = () => {
+		const context = canvas.getContext('2d');
+		canvas.width = state.width;
+		canvas.height = state.height;
+		context.drawImage(video, 0, 0, state.width, state.height)
+		const img = context.getImageData(0, 0, canvas.width, canvas.height)
+
+		const parent = currentScript.parentElement
+		parent.value = [img.width, img.height, ...img.data]
```

Comment:
If we would send data as a Uint8Array instead of a JS Array, it would be much much faster! How about:
```js
{
width: img.width,
height: img.height,
data: img.data,
}
```